store/src/java/com/zimbra/cs/datasource/DataSourceManager.java

```diff
@@ -216,6 +216,23 @@ public DataImport getDataImport(DataSource ds, boolean test) throws ServiceExcep
             } catch (Exception x) {
                 ZimbraLog.datasource.warn("Failed instantiating xsync class: %s", ds, x);
             }
+        case custom:
+            try {
+                String className = ds.getDataSourceImportClassName();
+                if (className != null && className.length() > 0) {
+                    Class<?> cmdClass;
+                    try {
+                        cmdClass = Class.forName(className);
+                    } catch (ClassNotFoundException x) {
+                        cmdClass = ExtensionUtil.findClass(className);
+                    }
+                    Constructor<?> constructor = cmdClass.getConstructor(new Class[] {DataSource.class});
```

Comment:
Worth having some error handling here in case cmdClass is still null at this point?